### PR TITLE
fixed grammar typos

### PR DESCRIPTION
**Pull request by tiagohermano**

### CODE REVIEW

File: chapters/ch06.asciidoc [hunk 1]
```diff
@@ -268,13 +268,13 @@ Proper integration testing might catch many of these kinds of mistakes, but that
...
+When we call into the backing API every time, with the full query, appeasing its quirks and shortcomings instead of taking the abstracted approach, it will not only be difficult to maintain an application that consumes those endpoints in more than one place, but it will also become a challenge down the road, when we want to include results from a different provider, which of course would have their own set of quirks and shortcomings. At this point we would have two separate sets of API calls, one for each provider, and each massaging the data to accommodate provider-specific quirks in a module which shouldn't be concerned with such matters, but only the results themselves.
```
Comment (tiagohermano) [inline chapters/ch06.asciidoc:277]: fixed 'accomodate' to 'accommodate'

File: chapters/ch06.asciidoc [hunk 2]
```diff
@@ -268,13 +268,13 @@ Proper integration testing might catch many of these kinds of mistakes, but that
...
 
-This intermediate layer would be in charge of calling the complex abstraction itself, but offers a simpler interface with less configuration options and improved ease of use for the use cases that matter to us. Often, complicated or legacy interfaces demand that we offer up data that could be derived from other parameters being passed into the function call. For example, we might be asked how many adults, how many children, and how many people in total are looking to make a flight booking, even though the latter can be derived from the former. Other examples include expecting fields to be in a particular string format (such as a date string that could be derived from a native JavaScript date instead), using nomenclature that's relevant to the implmentation but not so much to the consumer, or a lack of sensible defaults (required fields which are rarely changed into anything other than a recommended value that isn't set by default).
```
Comment (tiagohermano) [inline chapters/ch06.asciidoc]: fixed 'implmentation' to 'implementation'

File: chapters/ch06.asciidoc [hunk 3]
```diff
@@ -268,13 +268,13 @@ Proper integration testing might catch many of these kinds of mistakes, but that
...
+This intermediate layer would be in charge of calling the complex abstraction itself, but offers a simpler interface with less configuration options and improved ease of use for the use cases that matter to us. Often, complicated or legacy interfaces demand that we offer up data that could be derived from other parameters being passed into the function call. For example, we might be asked how many adults, how many children, and how many people in total are looking to make a flight booking, even though the latter can be derived from the former. Other examples include expecting fields to be in a particular string format (such as a date string that could be derived from a native JavaScript date instead), using nomenclature that's relevant to the implementation but not so much to the consumer, or a lack of sensible defaults (required fields which are rarely changed into anything other than a recommended value that isn't set by default).
```
Comment (tiagohermano) [inline chapters/ch06.asciidoc:271]: fixed 'implmentation' to 'implementation'

File: chapters/ch06.asciidoc [hunk 4]
```diff
@@ -219,7 +219,7 @@ State is almost ubiquitous, and practically a synonym of applications, because a
...
+Derived state in the form of caches is not uncommon in the world of web servers. In the case of a personal website with some books available for download, for instance, we might be tempted to store the PDF representation of each book in a file, so that we don't have to recompile the PDF whenever the corresponding `/book` route is visited. When the book is updated, we'd recompute the PDF file and flush it to disk again, so that this derived state remains fresh. When our web server ceases to be single-node and we start using a cluster of several nodes, however, it might not be so trivial to broadcast the news about books being updated across nodes, and thus it'd be best to leave derived state to the persistance layer. Otherwise, a web server node might receive the request to update a book, perform the update and recompute the PDF file on that node, but we’d be failing to invalidate the PDF files being served by other nodes, which would have, and continue to serve stale copies of the PDF representation.
```
Comment (tiagohermano) [inline chapters/ch06.asciidoc:222]: fixed 'uncmoon' to 'uncommon'

**Review by bevacqua — APPROVED** [review]

**Pull request merged**